### PR TITLE
roachtest: changes to reuse spot VM clusters

### PR DESCRIPTION
**Pull request by nameisbhaskar**
We have a provision to use spot VMs for running tests by setting a flag. This flag is set when the test is selected for running, after a cluster is selected. This ends up creating a new cluster with spot VM every time a test is marked to use spot VM. The change is to set the testspec to use spot Vm before the tests gets registered to the worker.

Fixes: #116778
Epic: none

### CODE REVIEW

**Comment by srosenberg** [comment]
Can we do a full run to get a rough idea on the frequency of preemptions after enabling reuse?

File: pkg/cmd/roachtest/test_runner.go [hunk 1]
```diff
@@ -328,7 +328,21 @@ func (r *testRunner) Run(
 		// Don't spin up more workers than necessary.
 		parallelism = n * count
 	}
+	for _, testSpec := range tests {
+		//  TODO(babusrithar): remove this once we see enough data in
+		//  nightly runs. This is a temp logic to test spot VMs.
```
Comment (renatolabs) [inline pkg/cmd/roachtest/test_runner.go:333]: FWIW, this comment is largely inaccurate at this point. We already tested spot VMs (and several subsequent PRs went in to make spot VM usage smoother for us and other teams). If anything, the TODO here should be to use spot VMs in a more useful way, especially in combination with test selection.

**Comment by renatolabs** [comment]
> Can we do a full run to get a rough idea on the frequency of preemptions after enabling reuse?

Another option is to merge it and use the regular nightly build on master the following night to learn a more realistic approximation of the number of preemptions; we can always revert the next day.

Number of preemptions can vary a lot depending on time of day, so running at the same time as the nightly will probably provide us with a better idea of the impact.

**Review by renatolabs — APPROVED** [review]

File: pkg/cmd/roachtest/test_runner.go (hunk 1 above)
Comment (nameisbhaskar) [reply pkg/cmd/roachtest/test_runner.go:333]: Thanks @renatolabs fro the review! I just copy pasted the old code to here :) - will change this.

**Comment by nameisbhaskar** [comment]
Thanks @renatolabs @srosenberg for your reviews. I will merge the changes when the CI passes.

**Comment by nameisbhaskar** [comment]
bors r=renatolabs,srosenberg

**Pull request merged**